### ISSUE TITLE
Qdrant Document object not correct

### ISSUE THREAD

**Issue by jrinder42** [hwchase17/langchain#5473]
### System Info

Langchain Version: 0.0.186
MacOS Ventura 13.3 - M1
Python 3.10.8

### Who can help?

@hwchase17 
@agola11 

### Information

- [X] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [X] Document Loaders
- [X] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

There is an error in the Qdrant Vectorstore code ([`qdrant.py`](https://github.com/hwchase17/langchain/blob/master/langchain/vectorstores/qdrant.py)). Specifically, with the function `_document_from_scored_point` on line 468 of [`qdrant.py`](https://github.com/hwchase17/langchain/blob/master/langchain/vectorstores/qdrant.py). The Document object is from the [`schema.py`](https://github.com/hwchase17/langchain/blob/master/langchain/schema.py) The function takes as few arguments:

page_content: str
metadata: dict = Field(default_factory=dict)

The [`qdrant.py`](https://github.com/hwchase17/langchain/blob/master/langchain/vectorstores/qdrant.py) file incorrectly makes the *metadata* parameter a string instead of a dict. 

This creates a few problems: 
1. If the *metadata* parameter in the function `_document_from_scored_point` is passed anything but None or a key that is not in the score_point object (a.k.a. None), it will error out. This is because this variable should be a dict, which is not returned from a dictionary *get* method.
2. The *metadata_payload_key* parameter does not seem to have a purpose / does not make sense given the above context.
 3. It is impossible for metadata to be returned when using the Qdrant *similarity_search* function within Langchain due to this issue.

### Expected behavior

I would like to be able to return metadata when using similarity_search with Qdrant. If you run [this](https://www.pinecone.io/learn/langchain-retrieval-augmentation/) example / focus on the vectorstore part and swap out the Pinecone work for Qdrant, there does not seem to be a way to use similarity search with metadata similar to how the example shows it.

**Comment by kacperlukawski** [comment]
@jrinder42 I'd appreciate a code snippet for reproducing the issue. I could reproduce Pinecone's example with Qdrant being the vector store to some extent. I wonder why they put the data directly to the index instead of relying on Langchain methods. That would make it more convenient. 

In the meantime, let me describe the purpose of the `metadata_payload_key` and `content_payload_key` params. When you put any documents or texts using `Qdrant.from_texts` or `qdrant.add_texts`, they must be represented somehow in the Qdrant payload. From the very beginning, we were representing each instance of `Document` by a JSON object like this:

```json
{
    "page_content": "Lorem ipsum",
    "metadata": {
        "key_1": "value_1",
        "key_2": 2,
        ...
    }
}
```

That was derived from the fields of a `Document`, but some people wanted to reuse existing collections. That's why we decided to introduce those two parameters:
-  `content_payload_key` - for the key used to store the page content
- `metadata_payload_key` - for the key used to store the metadata (we're not changing the type of metadata from `dict` to `str`, but allow specifying the key in the payload used to store the metadata)

Assuming you already had a Qdrant collection filled with some data, and you also kept the textual content used to generate the embeddings, you can simply reuse it in the Langchain app. For example, assume you create an instance of `Qdrant` in the following way:

```python
qdrant_client = QdrantClient(...)
vec_store = Qdrant(qdrant_client, ..., content_payload_key="my_text", metadata_payload_key="attributes")
```
That allows loading the data from a collection with such payloads:

```json
{
    "my_text": "Lorem ipsum",
    "attributes": {
        "key_1": "value_1",
        "key_2": 2,
        ...
    }
}
```

Moreover, if you don't have anything more except for the text, it it also possible to reuse the collection. It will work perfectly fine with the following payload:


```json
{
    "my_text": "Lorem ipsum"
}
```

**Comment by jrinder42** [comment]
Sure. I assumed that is how langchain thought the metadata was represented. However, that is not how metadata is represented in other vectorstores like Pinecone and Chroma. It is typically represented as 
```json
{"text": "main text", "metadata 1": "something", "metadata 2": "something else", etc}
```
Hence, it is not of the json / dictionary structure langchain is assuming it is in / not possible to get all of the metadata fields. I am simply pointing out how this structure assumption is different than other vectorstores in that other vectorestores store all text / metadata in a dictionary and you simply have to pluck out the main text from said dictionary. As far as I can tell, Qdrant should not be an exception to this rule. If it is an exception, it should be made clear of the assumed structure.

```python
client = QdrantClient()
# the below is in a loop over the data
# upsert
metadatas = {"meta 1": "something", "meta 2": "something", etc}
ids = list of ids
embeds = list of OpenAIEmbeddings().embed_documents(main texts from data)
client.upsert(collection_name='some name',
wait=True,
points=[PointStructure(id=elem[0], vector=elem[1], payload=elem[2]) for elem in zip(ids, embeds, metadatas)])
```
Are you saying that the metadatas needs to be structured like:
```json
{
    "metadata_payload_key": {
        "meta 1": "something",
        "meta 2": "something",
        etc
    }
}
```

My main question is why is a different payload structure assumed here relative to other vectorstores?

**Comment by kacperlukawski** [comment]
@jrinder42 As far as I recall, there was no consensus on how the metadata should look, as this is vendor-specific. In the case of Qdrant, we solved that by reflecting the structure of the `Document` so it's encoded as JSON, and we were one of the first vector DBs integrated into Langchain.

This is a rare case but may happen with the other engines since they decided to store both `page_content` and `metadata` attributes in a flat structure. Imagine you use `text` as a property for storing `Document.page_content`, and `Document.metadata` has the `text` property used for a different purpose. One of the properties will be gone since the attributes will be flattened. And which one of them will be lost depends on the vendor implementation.

I'm unsure about Chroma, but Pinecone accepts key-value as metadata only, and objects are not valid values. So they had no choice.

To sum it up, payloads are assumed to have the following structure:

```json
{
    "content_payload_key": "page content from the document",
    "metadata_payload_key": {
        "meta 1": "something",
        "meta 2": "something",
        etc
    }
}
```

I agree this should be well-described in the docs, and I'll improve the documentation to state that clearly.

**Comment by eric2788** [comment]
Meanwhile, langchain4j is deserializing the document as flattened, causing an error... could we have a unified standard between different languages 😢

**Comment by kacperlukawski** [comment]
@Anush008 Could you please have a look?

**Comment by anush008** [comment]
Hey @eric2788 and @kacperlukawski 

Langchain4j isn't a project maintained by the Langchain-AI team. It's from a different set of devs, IIRC.

So I don't think they're to be compatible with each other.

**Comment by eric2788** [comment]
> Hey @eric2788 and @kacperlukawski 
> 
> Langchain4j isn't a project maintained by the Langchain-AI team. It's from a different set of devs, IIRC.
> 
> So I don't think they're to be compatible with each other.

I see, I have reported the issue on both sides, hope the author of langchain4j would change for that
